client/src/lib/router/NavLink.tsx

```diff
@@ -35,14 +57,22 @@ export default function NavLink(props: NavLinkProps) {
   if (exact) {
     if (typeof to === 'string') {
       isActive = pathname === to;
-    } else if (to.pathname) {
-      isActive = pathname === to.pathname;
+    } else {
+      let match = true;
+      if (to.pathname && to.pathname !== pathname) {
+        match = false;
+      }
+      isActive = match && checkSearchAndHash(to, search, hash);
     }
   } else {
     if (typeof to === 'string') {
       isActive = pathname.startsWith(to);
-    } else if (to.pathname) {
-      isActive = pathname.startsWith(to.pathname);
+    } else {
+      let match = true;
+      if (to.pathname && !pathname.startsWith(to.pathname)) {
+        match = false;
+      }
```

Comment:
```suggestion
      const match = (!to.pathname || pathname.startsWith(to.pathname))
```

이렇게 한줄로 표현할 수 있을것 같습니다

Reply:
그럴 수 있겠네요 ㅎㅎ

Reply:
좋습니다~

Reply:
이게 맨처음에는 `checkSearchAndHash` 함수가 코드 안에 있어서 그랬었는데 합치면 되겠네요.

Reply:
수정했습니다!
```typescript
isActive = !(to.pathname && !pathname.startsWith(to.pathname)) && checkSearchAndHash(to, search, hash);
```
아예 한줄로 만들었습니다.

Reply:
멋져요!!